proto/ord_schema.proto

```diff
@@ -237,52 +237,69 @@ message CompoundIdentifier {
     // Protein data bank ID (for enzymes)
     PDB_ID = 13;
   }
-  IdentifierType type = 1;
-  // Custom compound identifier 
-  string type_custom = 2;
-  // Value of the compound identifier; certain types (e.g., pubchem_cid) may
+  oneof kind {
+    IdentifierType type = 1;
+    string custom = 2;
+  }
+  // Value of the compound identifier; certain types (e.g., PUBCHEM_CID) may
   // cast the string as an integer for downstream processing and validation.
   string value = 3;
 }
 
-message ReactionSetup {
-  enum VesselType {
-    VESSEL_TYPE_UNKNOWN = 0;
-    ROUND_BOTTOM_FLASK = 1;
-    VIAL = 2;  // Regular glass vial
-    WELL_PLATE = 3;
-    MICROWAVE_VIAL = 4;
-    TUBE = 5;
-    CONTINUOUS_STIRRED_TANK_REACTOR = 6;
-    PACKED_BED_REACTOR = 7;
-  }
-  VesselType vessel_type = 1;
-  // Used to describe an UNKNOWN vessel type or elaborate.
-  string vessel_type_custom = 2;
-  enum VesselMaterial {
-    VESSEL_MATERIAL_UNKNOWN = 0;
-    GLASS = 1;
-    POLYPROPYLENE = 2;
-    PLASTIC = 3;
-  }
-  VesselMaterial vessel_material = 3;
-  string vessel_material_custom = 4;
-  enum VesselPrep {
-    VESSEL_PREP_UNKNOWN = 0;
-    NONE = 1;
-    OVEN_DRIED = 2;
+message Vessel {
+  message VesselForm {
+    enum VesselFormType {
+      UNSPECIFIED = 0;
+      ROUND_BOTTOM_FLASK = 1;
+      VIAL = 2;
+      WELL_PLATE = 3;
+      MICROWAVE_VIAL = 4;
+      TUBE = 5;
+      CONTINUOUS_STIRRED_TANK_REACTOR = 6;
+      PACKED_BED_REACTOR = 7;
+    }
+    oneof kind {
+      VesselFormType type = 1;
+      string custom = 2;
+    }
   }
-  VesselPrep vessel_prep = 5;
-  // Used to describe an UNKNOWN vessel prep.
-  string vessel_prep_custom = 6;
-  // Size (volume) of the vessel.
-  Volume vessel_volume = 7;
+  message VesselMaterial {
+    enum VesselMaterialType {
+      UNSPECIFIED = 0;
+      GLASS = 1;
+      POLYPROPYLENE = 2;
+      PLASTIC = 3;
+    }
+    oneof kind {
+      VesselMaterial type = 1;
+      string custom = 2;
+    }
+  }
+  message VesselPrep {
```

Comment:
"Prep" or "Preparation"? The latter is consistent with what we use for Compounds.

Reply:
In general avoiding abbreviations is more style-compliant, so let's go with "Preparation".

Reply:
Changed:
* `Preparation` -> `CompoundPreparation`
* `VesselPrep{Type}` -> `VesselPreparation{Type}`